components/ChatSideBar.tsx

```diff
@@ -0,0 +1,142 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { useRouter } from "next/navigation";
+import { MessageSquare, X, Trash2 } from "lucide-react";
+
+interface Chat {
+  id: number;
+  name: string;
+  createdAt: string;
+}
+
+interface ChatSidebarProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export function ChatSidebar({ isOpen, onClose }: ChatSidebarProps) {
+  const [chats, setChats] = useState<Chat[]>([]);
```

Comment:
The Chat History displays blank area misleading the user to convey user has no history.

You should display a spinner when you fetch data, to handle that in state, you could either create `isLoading, setIsLoading` or better, create a unified state consisting of discriminated union such as `{type: 'ready', history: Chat[]} | {type: 'loading'} | {type: 'error', message: "..."}` And handle these states accordingly.